profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.admin.inc

```diff
@@ -0,0 +1,597 @@
+<?php
+/**
+ * @file
+ * Nexteuropa Newsroom administration page.
+ */
+
+/**
+ * Nexteuropa newsroom admin form.
+ */
+function nexteuropa_newsroom_admin_settings($form, &$form_state) {
+  // Be aware about the selected content types in the fields fieldset.
+  $sel_ct = NewsroomHelper::getFieldInstances('select');
```

Comment:
I have updated the code and used "field_info_field_map", which gave us small improvement.

Previously:
Total: 0.0012030601501465
Total: 0.00046205520629883

Now:
Total: 3.7908554077148E-5
Total: 1.9073486328125E-5

Again, this code is used only in the admin part.